.github/workflows/tests.yml

```diff
@@ -27,14 +27,10 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install -e .[dev]
-    
-    - name: Check formatting with black
-      run: |
-        black --check sbi tests
-    
-    - name: Check imports with isort
+
+    - name: pre-commit hooks, ruff linting and formatting
       run: |
-        isort -c sbi tests
+        pre-commit run --all-files
```

Comment:
Instead of running pre-commit manually, you could use the [pre-commit/action](https://github.com/pre-commit/action). This action automatically caches the pre-commit dependencies, which makes CI faster. See [example](https://github.com/probabilists/zuko/blob/master/.github/workflows/ci.yml) from Zuko's CI.

Reply:
thanks for the pointer! 🙏